GitUI/UserControls/FileStatusList.cs

```diff
@@ -924,19 +927,21 @@ private void UpdateFileStatusListView(bool updateCausedByFilter = false)
                         listItem.ImageIndex = GetItemImageIndex(item);
                     }
 
-                    if (item.GetSubmoduleStatusAsync() is not null && !item.GetSubmoduleStatusAsync().IsCompleted)
+                    if (item.IsSubmodule
+                        && item.GetSubmoduleStatusAsync() is Task<GitSubmoduleStatus> task
+                        && task is not null)
```

Comment:
@gerhardol the type check is essentially a null check too.

```suggestion
                        && item.GetSubmoduleStatusAsync() is Task<GitSubmoduleStatus> task)
```

Reply:
Another way of writing this is:

```suggestion
                        && item.GetSubmoduleStatusAsync() is { } task)
```

But is that actually better than assigning the task to a local and doing an explicit `is not null` test? The only downside is a bit of nesting, which is not the end of the world.

Reply:
We had this discussion before: This is a hard to read. One does not need to use every new C# syntax invention.

Reply:
The `task is not null` checks in FileStatusList.cs are not needed.
I changed to that style as it is required in FormCommit

I prefer to have a minimal code why my preference is to just removing is not null where possible.